tools/kagami/src/genesis.rs

```diff
@@ -241,10 +259,231 @@ fn generate_synthetic(
                     ),
                 )
                 .into();
-                first_transaction.append_instruction(mint);
+                first_transaction.push_instruction(mint);
             }
         }
     }
 
     Ok(genesis)
 }
+
+/// Use `Kagami` to sign genesis block.
+#[derive(Clone, Debug, Parser)]
+#[command(group = ArgGroup::new("private_key").required(true))]
+#[command(group = ArgGroup::new("public_key").required(true))]
+pub struct SignArgs {
+    /// Path to genesis json file
+    #[clap(long)]
+    genesis_file: PathBuf,
+    /// The algorithm of the provided keypair
+    #[clap(default_value_t, long, short)]
+    algorithm: crypto::AlgorithmArg,
+    /// Private key (in hex string format) to sign genesis block
+    #[clap(long, group = "private_key")]
+    private_key_string: Option<String>,
+    /// Path to private key to sign genesis block
+    #[clap(long, group = "private_key")]
+    private_key_file: Option<PathBuf>,
+    /// Public key in multihash format of the corresponding private key
+    #[clap(long, group = "public_key")]
+    public_key_string: Option<PublicKey>,
+    /// Path to public key in multihash format of the corresponding private key
+    #[clap(long, group = "public_key")]
+    public_key_file: Option<PathBuf>,
+    /// Path to json-serialized keypair
+    #[clap(long, short, group = "private_key", group = "public_key")]
+    keypair_file: Option<PathBuf>,
+    /// Unique id of blockchain
+    #[clap(long)]
+    chain_id: ChainId,
+    /// Encode signed genesis block with SCALE (it is only supported with file output)
```

Comment:
- If it works only with file output, please enforce it with a clap requirement on `--out-file`
- Without file output, you can still print a hex
- Actually, what do we have `--out-file` for?.. The final purpose is to produce a hex string which user will put into the `genesis.signature` config field (or `GENESIS_SIGNATURE` env). I don't see a purpose of having a binary file output. What do I miss?